File: C++/PubUtilLib/plScene/plSceneNode.cpp
#include "plSceneNode.h"

plSceneNode::plSceneNode(PlasmaVer pv) { }
plSceneNode::~plSceneNode() { }

void plSceneNode::read(hsStream *S) {
    hsKeyedObject::read(S);

    SceneObjects.setSize(S->readInt());
    for (int i=0; i<SceneObjects.getSize(); i++)
        SceneObjects[i].readRef(S);
    OtherObjects.setSize(S->readInt());
    for (int i=0; i<OtherObjects.getSize(); i++)
        OtherObjects[i].readRef(S);
}

void plSceneNode::write(hsStream *S) {
    hsKeyedObject::write(S);

    S->writeInt(SceneObjects.getSize());
    for (int i=0; i<SceneObjects.getSize(); i++)
        SceneObjects[i].writeRef(S);
    S->writeInt(OtherObjects.getSize());
    for (int i=0; i<OtherObjects.getSize(); i++)
        OtherObjects[i].writeRef(S);
}

File: Java/Plasma/PubUtilLib/plScene/plSceneNode.java
package Plasma.PubUtilLib.plScene;

import java.io.IOException;
import Plasma.CoreLib.hsStream;
import Plasma.CoreLib.hsTArray;
import Plasma.NucleusLib.pnKeyedObject.*;

public class plSceneNode extends hsKeyedObject {
    public hsTArray<plKeyRef> SceneObjects = new hsTArray<plKeyRef>();
    public hsTArray<plKeyRef> OtherObjects = new hsTArray<plKeyRef>();

    public void read(hsStream S) throws IOException {
        super.read(S);
        SceneObjects.read(S);
        OtherObjects.read(S);
    }

    public void write(hsStream S) throws IOException {
        super.write(S);
        SceneObjects.write(S);
        OtherObjects.write(S);
    }
}

